src/main/kotlin/com/team4099/robot2023/subsystems/drivetrain/swervemodule/SwerveModuleIO.kt

```diff
@@ -63,102 +64,107 @@ interface SwerveModuleIO {
       }
-      table?.get("steeringSupplyCurrentAmps", steeringSupplyCurrent.inAmperes)?.let {
-        steeringSupplyCurrent = it.amps
+      table?.get("supplyCurrentSteer", supplyCurrentSteer.inAmperes)?.let {
+        supplyCurrentSteer = it.amps
       }
-      table?.get("drivePositionMeters", drivePosition.inMeters)?.let { drivePosition = it.meters }
-      table?.get("steeringPositionRadians", steeringPosition.inRadians)?.let {
-        steeringPosition = it.radians
-      }
-      table?.get("driveVelocityMetersPerSecond", driveVelocity.inMetersPerSecond)?.let {
+      table?.get("steerTemp", steerTemp.inCelsius)?.let { steerTemp = it.celsius }
+      table?.get("driveTemp", driveTemp.inCelsius)?.let { driveTemp = it.celsius }
+      table?.get("drivePosition", drivePosition.inMeters)?.let { drivePosition = it.meters }
+      table?.get("steerPosition", steerPosition.inRadians)?.let { steerPosition = it.radians }
+      table?.get("driveVelocity", driveVelocity.inMetersPerSecond)?.let {
         driveVelocity = it.meters.perSecond
       }
-      table?.get("steeringVelocityRadiansPerSecond", steeringVelocity.inRadiansPerSecond)?.let {
-        steeringVelocity = it.radians.perSecond
+      table?.get("steerVelocity", steerVelocity.inRadiansPerSecond)?.let {
+        steerVelocity = it.radians.perSecond
       }
-      table?.get("driveTempCelcius", driveTemp.inCelsius)?.let { driveTemp = it.celsius }
-      table?.get("steeringTempCelcius", steeringTemp.inCelsius)?.let { steeringTemp = it.celsius }
+      table?.get("drift", drift.inMeters)?.let { drift = it.meters }
       table?.get("potentiometerOutputRaw", potentiometerOutputRaw)?.let {
         potentiometerOutputRaw = it
       }
       table?.get("potentiometerOutputRaw", potentiometerOutputRadians.inRadians)?.let {
         potentiometerOutputRadians = it.radians
       }
-      table?.get("driftPositionMeters", drift.inMeters)?.let { drift = it.meters }
     }
   }
-
   val label: String
 
   fun updateInputs(inputs: SwerveModuleIOInputs) {}
 
-  fun setSteeringSetpoint(angle: Angle) {}
-  fun setClosedLoop(steering: Angle, speed: LinearVelocity, acceleration: LinearAcceleration) {}
   fun setOpenLoop(steering: Angle, speed: LinearVelocity) {}
-  fun runCharacterization(input: ElectricalPotential) {}
+
+  fun setClosedLoop(angle: Angle, speed: LinearVelocity, acceleration: LinearAcceleration) {}
 
   fun resetModuleZero() {}
-  fun zeroSteering(isInAutonomous: Boolean = false) {}
+
+  fun zeroSteering(isInAuto: Boolean = false) {}
+
   fun zeroDrive() {}
 
   fun setDriveBrakeMode(brake: Boolean) {}
 
   fun setSteeringBrakeMode(brake: Boolean) {}
 
-  fun configureDrivePID(
+  fun runCharacterization(input: ElectricalPotential) {}
+
+  fun setSteeringSetpoint(angle: Angle) {}
+
+  fun configSteerPID(
+    kP: ProportionalGain<Radian, Volt>,
+    kI: IntegralGain<Radian, Volt>,
+    kD: DerivativeGain<Radian, Volt>
+  ) {}
+
+  fun configDrivePID(
```

Comment:
can we separate FF config from PID? not the same thing

Reply:
i did but i have a few concerns i posted in sbt_programming regarding the SwerveModuleIOTalon implementation

Reply:
resolved!